builder/solver_monitor.go

```diff
@@ -316,11 +317,12 @@ Loop:
 		}
 	}
 	if errVertex != nil {
-		sm.reprintFailure(errVertex)
+		sm.reprintFailure(errVertex, phaseText)
 	}
 	sm.mu.Lock()
-	if sm.success {
-		sm.console.PrintSuccess()
+	if sm.success && !sm.printedSuccess {
+		sm.console.PrintSuccess(phaseText)
+		sm.printedSuccess = true
```

Comment:
Should this also get `sm.lastOutputWasOngoingProgress = false`?

Reply:
Probably. Most cases we are ending up printing through SetSuccess now; so unlikely to get here first.